src/oc_erchef/apps/data_collector/src/data_collector_http.erl

```diff
@@ -0,0 +1,85 @@
+-module(data_collector_http).
+
+-export([
+         request/3,
+         request/4,
+         get/1,
+         get/2,
+         get/3,
+         post/2,
+         post/3,
+         delete/2,
+         delete/3,
+         create_pool/0,
+         delete_pool/0
+        ]).
+
+-define(DEFAULT_HEADERS, [{"Content-Type", "application/json"}]).
+
+request(Path, Method, Body) ->
+    request(Path, Method, Body, ?DEFAULT_HEADERS).
+
+request(Path, Method, Body, Headers) ->
+    {ok, Timeout} = application:get_env(data_collector, timeout),
+    oc_httpc:request(?MODULE, Path, Headers, Method, Body, Timeout).
+
+%%
+%% Simple helpers for requets when you only
+%% care about success or failure.
+%%
+-spec get(list()) -> ok | {error, term()}.
+get(Path) ->
+    get(Path, [], ?DEFAULT_HEADERS).
+
+-spec get(list(), iolist() | binary()) -> ok | {error, term()}.
+get(Path, Body) ->
+    get(Path, Body, ?DEFAULT_HEADERS).
+
+-spec get(list(), iolist() | binary(), list()) -> ok | {error, term()}.
+get(Path, Body, Headers) ->
+    request_with_caught_errors(Path, get, Body, Headers).
+
+-spec post(list(), iolist() | binary()) -> ok | {error, term()}.
+post(Path, Body) ->
+    post(Path, Body, ?DEFAULT_HEADERS).
+
+-spec post(list(), iolist() | binary(), list()) -> ok | {error, term()}.
+post(Path, Body, Headers) ->
+    request_with_caught_errors(Path, post, Body, Headers).
+
+-spec delete(list(), iolist() | binary()) -> ok | {error, term()}.
+delete(Path, Body) ->
+    delete(Path, Body, ?DEFAULT_HEADERS).
+
+-spec delete(list(), iolist() | binary(), list()) -> ok | {error, term()}.
+delete(Path, Body, Headers) ->
+    request_with_caught_errors(Path, delete, Body, Headers).
+
+request_with_caught_errors(Path, Method, Body, Headers) when is_list(Body)->
+    request_with_caught_errors(Path, Method, iolist_to_binary(Body), Headers);
+request_with_caught_errors(Path, Method, Body, Headers) ->
+    try
+        case request(Path, Method, Body, Headers) of
+            {ok, "200", _Head, _RespBody} -> ok;
```

Comment:
Chef's implementation of the Data Collector server currently returns a `204 No Content` when we post the payload. We probably need to account for that and perhaps consider any 2xx status code as OK.